docs/julia/Runner.md

```diff
@@ -0,0 +1,119 @@
+
+A module doesn't have to support all of the objective types. If a user asks the runner to benchmark an unsupported objective, the runner prints an error and stops.
+
+Every module, for every objective it supports, exports a parameterless factory function that returns an object of the corresponding instantiation of generic [`Test{Input, Output}`](./Modules.md#itest-implementation) type.
+
+## Supported Objective Types
+Currently supported objective types:
+     
+| Full Name | Short Name |
+| -- | -- |
+| Gaussian Mixture Model Fitting | GMM |
+| Bundle Adjustment| BA |
+| Hand Tracking | Hand |
+| Long short-term memory | LSTM |
+
+
+## Command Line
+
+```powershell
+julia --project=/path/to/repo/JuliaProject.toml --optimize=3 Runner.jl test_type module_path input_filepath output_dir minimum_measurable_time nruns_F nruns_J time_limit [-rep]
+```
+
+ - `test_type` - The short type name of the objective function. It may also be equal to "Hand-Complicated" that designates the complicated case of the "Hand" objective, where the variable U is considered (see [srajer-autodiff-screen.pdf](../../Documents/srajer-autodiff-screen.pdf), page 5).
+ - `module_path` - an absolute or relative path to a `.jl` file containing the module to be benchmarked.
+ - `input_filepath` - an absolute or relative path to the input file.
+ - `output_dir` - a directory where the output files should be stored.
+ - `minimum_measurable_time` - minimum time that the computation needs to run to produce a reliable measurement.
+ - `nruns_F` - maximum number of times to run the computation of the objective function for timing.
+ - `nruns_J` - maximum number of times to run the computation of the considered derivative (gradient or Jacobian) for timing.
+ - `time_limit` - soft (see [Methodology.md](../Methodology.md)) time limit for benchmarking the computation of either the objective function or its gradient/Jacobian.
+ - `-rep` *(applicable only for GMM)* - if enabled, all input data points are expected to have one shared value.
+
+## Adding new modules
+(see [Modules.md](./Modules.md))
+
+## Adding new objective types
+
+1. Define a module `TData` in the `/src/julia/shared/TData.jl` file where `T` is the name of the new objective. In that module define and export the following
+
+    ```julia
+    struct TInput
+    ```
+    The data type for the new objective's inputs.
+    ```julia
+    mutable struct TOutput
+    ```
+    A mutable structure that can hold outputs of both new objective function and its target Jacobian.
+    ```julia
+    function empty_t_output()::TOutput
+    ```
+    A function that returns an empty object of the `TOutput` type, that later will be populated by the benchmarked module.
+    ```julia
+    function load_t_input(fn::AbstractString)::TInput
+    ```
+    Open input file `fn` and loads data to the structure of the TInput type.
```

Comment:
Keeping a style, "TInput" should be inside \`\`

Reply:
Fixed.